apps/site/src/app/apply/sections/Form/Form.tsx

```diff
@@ -8,20 +13,82 @@ import ResumeInformation from "./ResumeInformation";
 
 import styles from "./Form.module.scss";
 
+const APPLY_PATH = "/api/user/apply";
+const FIELDS_WITH_OTHER = ["pronouns", "ethnicity", "school", "major"];
+
 export default function Form() {
+	const [submitting, setSubmitting] = useState(false);
+	const [sessionExpired, setSessionExpired] = useState(false);
+
+	const handleSubmit = async (
+		event: FormEvent<HTMLFormElement>,
+	): Promise<void> => {
+		// Disable native post submission
+		event.preventDefault();
+		setSubmitting(true);
+		setSessionExpired(false);
+
+		const formData = new FormData(event.currentTarget);
+
+		// Use other values when selected
+		for (const field of FIELDS_WITH_OTHER) {
+			const otherField = `other_${field}`;
+			if (formData.get(field) === "other") {
+				formData.set(field, formData.get(otherField)!);
+				formData.delete(otherField);
+			}
+		}
+
+		const formEntries = Object.fromEntries(formData.entries());
+		console.debug(formEntries);
+
+		try {
+			const res = await axios.post(APPLY_PATH, formData);
+			if (res.status === 201) {
+				console.log("Application submitted");
+				// TODO: show submitted message
+			}
+		} catch (err) {
+			console.error(err);
+			if (axios.isAxiosError(err)) {
+				if (err.response?.status === 401) {
+					setSessionExpired(true);
+				}
+			}
+		}
+
+		setSubmitting(false);
+	};
+
+	const sessionExpiredMessage = (
+		<p className="text-red-500">
+			Your session has expired. Please{" "}
+			<a
+				href="/login"
+				target="_blank"
+				className="text-blue-600 underline"
+			>
+				log in from a new tab
+			</a>{" "}
+			to restore your session and then try submitting again.
+		</p>
+	);
```

Comment:
Issue: We'll want to add some margin on the sides of this element on mobile views.

<img width="350" alt="image" src="https://github.com/HackAtUCI/irvinehacks-site-2024/assets/28026520/daee1fb1-9cce-4913-a095-2fbdef328cb1">

Reply:
Looks like we're gonna need a container design overhaul but added for now in 196d33f.